kafka-connect-cloud-common/src/main/scala/io/lenses/streamreactor/connect/cloud/common/formats/reader/converters/SchemaAndValueEnvelopeConverter.scala

```diff
@@ -80,13 +81,15 @@ class SchemaAndValueEnvelopeConverter(
     if (fields.contains("key")) {
       key       = struct.get("key")
       keySchema = struct.schema().field("key").schema()
+      key       = byteBufferToArray(keySchema, key)
     }
```

Comment:
Do you think it would be better if we could avoid the mutable var state and also overwriting the same value immediately.